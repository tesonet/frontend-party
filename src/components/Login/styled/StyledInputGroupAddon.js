import React from 'react';
import styled from 'styled-components';
import { InputGroupAddon } from 'reactstrap';

const StyledContainer = styled(InputGroupAddon)`
  background-color: ${props => props.theme.bgColorPrimary};
  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;
  padding-left: 1.5rem;
  padding-right: 0;

  & > .icon {
    fill: ${props => props.theme.inputTextColorPrimary};
    height: 3rem;
  }
`;

const StyledInputGroupAddon = props => <StyledContainer addonType="prepend" {...props} />;

export default StyledInputGroupAddon;
