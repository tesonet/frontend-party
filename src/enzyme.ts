import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
