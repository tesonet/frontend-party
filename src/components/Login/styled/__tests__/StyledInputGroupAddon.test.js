import React from 'react';
import { shallow } from 'enzyme';
import { InputGroupAddon } from 'reactstrap';

import StyledInputGroupAddon from '../StyledInputGroupAddon';

describe('StyledInputGroupAddon component', () => {
  it('should render the <InputGroupAddon /> element.', () => {
    const renderedComponent = shallow(<StyledInputGroupAddon />).dive();

    expect(renderedComponent.type()).toEqual(InputGroupAddon);
    expect(renderedComponent).toMatchSnapshot();
  });
});
