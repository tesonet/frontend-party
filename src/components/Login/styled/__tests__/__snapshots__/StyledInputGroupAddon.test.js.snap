// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInputGroupAddon component should render the <InputGroupAddon /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(InputGroupAddon)
    addonType="prepend"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "addonType": "prepend",
      "className": "sc-bdVaJa cJVzTl",
      "tag": "div",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addonType": "prepend",
        "className": "sc-bdVaJa cJVzTl",
        "tag": "div",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
